Library/Homebrew/utils/spdx.rb

```diff
@@ -5,19 +5,111 @@
 module SPDX
   module_function
 
-  JSON_PATH = (HOMEBREW_LIBRARY_PATH/"data/spdx.json").freeze
+  JSON_PATH = (HOMEBREW_LIBRARY_PATH/"data").freeze
   API_URL = "https://api.github.com/repos/spdx/license-list-data/releases/latest"
 
-  def spdx_data
-    @spdx_data ||= JSON.parse(JSON_PATH.read)
+  def license_data
+    @license_data ||= JSON.parse (JSON_PATH/"spdx_licenses.json").read
+  end
+
+  def exception_data
+    @exception_data ||= JSON.parse (JSON_PATH/"spdx_exceptions.json").read
   end
 
   def latest_tag
     @latest_tag ||= GitHub.open_api(API_URL)["tag_name"]
   end
 
   def download_latest_license_data!(to: JSON_PATH)
-    data_url = "https://raw.githubusercontent.com/spdx/license-list-data/#{latest_tag}/json/licenses.json"
-    curl_download(data_url, to: to, partial: false)
+    data_url = "https://raw.githubusercontent.com/spdx/license-list-data/#{latest_tag}/json/"
+    curl_download(data_url + "licenses.json", to: to/"spdx_licenses.json", partial: false)
+    curl_download(data_url + "exceptions.json", to: to/"spdx_exceptions.json", partial: false)
+  end
+
+  def parse_license_expression(license_expression)
+    licenses = []
+    exceptions = []
+
+    case license_expression
+    when String, Symbol
+      licenses.push license_expression
+    when Hash
+      key = license_expression.keys.first
+      if [:any_of, :all_of].include? key
+        sub_license, sub_exception = parse_license_expression license_expression[key]
+        licenses += sub_license
+        exceptions += sub_exception
```

Comment:
I wonder if there's a way to share this code with 47-50 below 🤔

Reply:
Possibly. It might end up being more confusing to have them share code, but I'll play around with it.

Reply:
I think it is better to have the duplicate code here. I think it would hurt readability to try to remove it, but I'm open to suggestions